File: app/src/main/java/com/aquamorph/frcmanager/models/Rank.java
package com.aquamorph.frcmanager.models;

/**
 * Stores rank information at an event.
 *
 * @author Christian Colglazier
 * @version 12/27/2017
 */
public class Rank {
	public Rankings[] rankings = new Rankings[0];
	public ExtraStatsInfo[] extra_stats_info = new ExtraStatsInfo[0];
	public SortOrderInfo[] sort_order_info = new SortOrderInfo[0];

	public class Rankings {
		public int dq;
		public int matches_played;
		public double qual_average;
		public int rank;
		public WLTRecord record;
		public double[] extra_stats;
		public double[] sort_orders;
		public String team_key;
	}

	public class WLTRecord {
		public int losses;
		public int wins;
		public int ties;
	}

	public class ExtraStatsInfo {
		public String name;
		public int precision;
	}

	public class SortOrderInfo {
		public String name;
		public int precision;
	}
}

File: app/src/main/kotlin/com/aquamorph/frcmanager/models/Rank.kt
package com.aquamorph.frcmanager.models

import java.util.Locale

/**
 * Stores rank information at an event.
 *
 * @author Christian Colglazier
 * @version 1/30/2020
 */
data class Rank(
    var rankings: ArrayList<Rankings>,
    var extra_stats_info: ArrayList<ExtraStatsInfo>,
    var sort_order_info: ArrayList<SortOrderInfo>
) {
    inner class Rankings(
        var dq: Int,
        var matches_played: Int,
        var qual_average: Double,
        var rank: Int,
        var record: WLTRecord,
        var extra_stats: DoubleArray,
        var sort_orders: DoubleArray,
        var team_key: String
    )

    data class WLTRecord(
        var losses: Int,
        var wins: Int,
        var ties: Int
    )

    data class ExtraStatsInfo(
        var name: String,
        var precision: Int
    )

    data class SortOrderInfo(
        var name: String,
        var precision: Int
    )

    companion object {
        fun recordToString(record: WLTRecord): String {
            return String.format(Locale.ENGLISH, "%d-%d-%d",
                    record.wins, record.losses, record.ties)
        }
    }
}
